src/controllers/DynamicFacetRequestBuilder.ts

```diff
@@ -0,0 +1,33 @@
+import { FacetType, IFacetRequest } from '../rest/Facet/FacetRequest';
+import { FacetSortCriteria } from '../rest/Facet/FacetSortCriteria';
+import { IQuery } from '../rest/Query';
+import { Utils } from '../utils/Utils';
+
+export interface IBasicFacetRequestParameters {
+  facetId: string;
+  field: string;
+  type: FacetType;
+  injectionDepth: number;
+  sortCriteria?: FacetSortCriteria;
+  delimitingCharacter?: string;
+  filterFacetCount?: boolean;
```

Comment:
Could instead this interface accepts a `DynamicFacet` kind of interface ? Meaning code using this module could more or less do : 
```
new DynamicFacetRequestBuilder(this.facet)
```

?

Reply:
Yes but that would take more work, we need to extract all the common properties for all facets. I might just do it for the interfaces for now.